sematic/resolvers/cloud_resolver.py

```diff
@@ -72,18 +90,62 @@ def set_graph(self, runs: List[Run], artifacts: List[Artifact], edges: List[Edge
         self._artifacts = {artifact.id: artifact for artifact in artifacts}
         self._edges = {make_edge_key(edge): edge for edge in edges}
 
-    def _get_resolution_image(self) -> Optional[str]:
-        return get_image_uri()
+    @memoized_property
+    def _resolution(self) -> Resolution:
+        return api_client.get_resolution(self._root_future.id)
+
+    @memoized_property
+    def _container_image_uris(self) -> Optional[Dict[str, str]]:
+        if self._detached:
+            return self._resolution.container_image_uris
+
+        return get_image_uris()
+
+    def _get_container_image(self, future: AbstractFuture) -> Optional[str]:
+        if self._container_image_uris is None:
+            return None
+
+        if future.props.inline and future is not self._root_future:
+            return self._runs[self._root_future.id].container_image_uri
```

Comment:
Under the assumption that the root run and the resolver are running with the same image, I don't understand the `and future is not self._root_future` clause in the conditional. I think returning the URI of the root run like this is exactly what you'd want in that case.